dapp/src/components/ProjectInfo.astro

```diff
@@ -0,0 +1,188 @@
+        }
+        const maintainersList = document.getElementById("project-maintainers");
+        if (maintainersList) {
+          const ul = maintainersList.querySelector("ul");
+          if (ul) {
+            ul.innerHTML = '';
+            projectInfo.maintainers.forEach((maintainer: string) => {
+              const li = document.createElement("li");
+              li.textContent = maintainer;
+              ul.appendChild(li);
+            });
+          }
+        }
+        const githubLink = document.getElementById("github-link") as HTMLAnchorElement;
+        if (githubLink) {
+          githubLink.href = `${projectInfo.config.url}`;
+        }
+
+        // Update modal content
+        const modalProjectName = document.getElementById("modal-project-name");
+        if (modalProjectName) {
+          modalProjectName.textContent = projectInfo.name;
+        }
+        
+        const modalMaintainers = document.getElementById("modal-maintainers") as HTMLInputElement;
+        if (modalMaintainers) {
+          modalMaintainers.value = projectInfo.maintainers.join(',');
+        }
+        const modalConfigUrl = document.getElementById("modal-config-url") as HTMLInputElement;
+        if (modalConfigUrl) {
+          modalConfigUrl.value = projectInfo.config.url;
+        }
+        const modalConfigHash = document.getElementById("modal-config-hash") as HTMLInputElement;
+        if (modalConfigHash) {
+          modalConfigHash.value = projectInfo.config.hash;
+        }
+
+        // Check if the connected wallet is a maintainer
+        const connectedPublicKey = loadedPublicKey();
+        const isMaintainer = connectedPublicKey ? projectInfo.maintainers.includes(connectedPublicKey) : false;
+        const updateConfigButton = document.getElementById("open-update-config-modal-button");
+        if (updateConfigButton) {
+          updateConfigButton.classList.toggle("hidden", !isMaintainer);
+        }
+      }
+    }
+
+    updateProjectInfo();
+
+    const button = document.querySelector(
+      "[update-config]",
+    ) as HTMLButtonElement; 
+    if (button) {
+      button.addEventListener("click", async () => {
+          console.log("Update config button clicked");
+          modal.showModal();
+      });
+    }
+
+    const updateButton = document.getElementById("update-config-button");
+    if (updateButton) {
+      updateButton.addEventListener("click", async () => {
+        const modalMaintainers = (document.getElementById("modal-maintainers") as HTMLInputElement).value;
+        const modalConfigUrl = (document.getElementById("modal-config-url") as HTMLInputElement).value;
+        const modalConfigHash = (document.getElementById("modal-config-hash") as HTMLInputElement).value;
+
+        const updateStatus = await updateConfig(
+          modalMaintainers,
+          modalConfigUrl,
+          modalConfigHash
+        );
+
+        if (updateStatus) {
+          updateProjectInfo();
```

Comment:
For a follow up, I think it should also reload the project info. And same as with the other buttons, it should add the spinner component so users know something happened after they clicked the button.